### PR TITLE
Week 36: Loan Calculator component

### PR DESCRIPTION
**Pull request by fespinoza**
# Why?

To give an option to offer loans for eiendom ads, week 36

# What?

- Add `LoanCalculatorView` with screenshot test
- Add `withAutoLayout` parameter to the `Button` initializer

# Show me

| Device | Loan Calculator | Error State |
| --- | --- | --- |
| **iPhone** | iphone | iphone-error |
| **iPad** | ipad  | ipad-error |

### CODE REVIEW

**Comment by fespinoza** [comment]
ready to review

File: Demo/Components/LoanCalculator/IntegerNumberSuffixFormatter.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    func string(for value: Float) -> String? {
+        return "\(super.string(from: value) ?? "") \(suffix)"
+    }
+}
+
+// MARK: - Extensions
+
+private extension NumberFormatter {
+    @objc func string(from value: Float) -> String? {
```
Comment (vadymmarkov) [inline Demo/Components/LoanCalculator/IntegerNumberSuffixFormatter.swift]: Why @objc?

Comment (fespinoza) [reply Demo/Components/LoanCalculator/IntegerNumberSuffixFormatter.swift]: No real reason, I removed it

File: Demo/Components/LoanCalculator/LoanCalculatorDemoView.swift [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+    private lazy var loanCalculatorView: LoanCalculatorView = {
+        let view = LoanCalculatorView(withAutoLayout: true)
+        view.dataSource = self
+        view.delegate = self
+        return view
+    }()
+
+    private lazy var currencyFormatter: IntegerNumberSuffixFormatter = IntegerNumberSuffixFormatter(suffix: "kr")
+    private lazy var yearsFormatter: IntegerNumberSuffixFormatter = IntegerNumberSuffixFormatter(suffix: "år")
```
Comment (bstien) [inline Demo/Components/LoanCalculator/LoanCalculatorDemoView.swift]: You could remove the type annotations for these two

File: Demo/Components/LoanCalculator/LoanCalculatorDemoView.swift [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    }
+
+    public required init?(coder aDecoder: NSCoder) { fatalError() }
+
+    private func setup() {
+        loanCalculatorView.configure(with: LoanCalculatorDemoViewModel.makeViewModel())
+        addSubview(loanCalculatorView)
+        NSLayoutConstraint.activate([
+            loanCalculatorView.widthAnchor.constraint(equalTo: widthAnchor, multiplier: 1, constant: -.largeSpacing),
+            loanCalculatorView.heightAnchor.constraint(lessThanOrEqualTo: heightAnchor, multiplier: 1, constant: -.largeSpacing),
```
Comment (bstien) [inline Demo/Components/LoanCalculator/LoanCalculatorDemoView.swift]: The `multiplier` parameter can be removed from these two lines

**Pull request merged**